packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -0,0 +1,191 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ChangeEncoder, ChangeFamily, JsonCompatibleReadOnly, ProgressiveEditBuilder } from "../../change-family";
+import { ChangeRebaser } from "../../rebase";
+import { FieldKindIdentifier } from "../../schema-stored";
+import { AnchorSet, Delta, FieldKey } from "../../tree";
+import { brand, getOrAddEmptyToMap } from "../../util";
+import { FieldChangeHandler, FieldChangeMap, FieldChange } from "./fieldChangeHandler";
+import { FullSchemaPolicy } from "./fieldKind";
+
+/**
+ * Implementation of ChangeFamily which delegates work in a given field to the appropriate FieldKind
+ * as determined by the schema.
+ */
+export class ModularChangeFamily implements
+    ChangeFamily<ModularEditBuilder, FieldChangeMap>,
+    ChangeRebaser<FieldChangeMap>,
+    ChangeEncoder<FieldChangeMap> {
```

Comment:
change encoder is a class not an interface, so "implementing it" is allowed but misleading (ex: people will expect instanceof to work and it won't, and you are missing out on the default method implementations, and changes to that class won't impact this code properly).

Probably better to not "return this" for the encoder and instead actually define one as is own object. Then you will bet binary format support for free, and the ability to test encoding without in a more targeted way.

Reply:
Thanks for catching that. It's annoying that Typescript doesn't complain about this.